### DISCUSSION TITLE
What authentication type for automated user management?

### DISCUSSION THREAD

**Discussion by leefrankeFNC** [community/community#22916] [New to GitHub]
<p>We want to automate user management in our organization and private repository.  We have an automated termination process and we want to add that when a person is termed from our company they are removed from our GitHub org and access removed to private repo.</p>
<p>I’ve been reading about the different types of authentication types that GitHub uses and have throughly confused myself as to which type I should use for this (FYI we have 2FA required).</p>
<p>At this point I’m thinking we should just create an ‘integraion account’ in GitHub with its own email address making it an owner of both, then create GitHub app bases on that user account to do the work.</p>
<p>Thanks,</p>
<p>lee</p>

**Comment by lee-dohm** [comment, marked as answer]
<p>There are probably multiple ways to do this but you could write a script that uses OAuth authentication to authenticate as the user running the script (assuming they had the appropriate access rights to remove organization members). You can see some of the code that I use for that kind of thing in the <a href="https://github.com/lee-dohm/dotfiles/blob/master/bin/github-labels#L57-L113" rel="nofollow noopener">scripts in my dotfiles repository</a>.</p>
<p>I hope that helps!</p>

**Comment by leefrankeFNC** [comment]
<p>I kept going back and forth which way to do it and the OAuth seems as good as any other.</p>
<p>Talk about some synchronicity. I had just thought about this question and started logging into GitHub when I got the email notification that you had replied.</p>
<p>Thanks for that help and code!</p>
<p>lee</p>
